packages/devextreme/js/__internal/core/r1/dom_component_wrapper.tsx

```diff
@@ -0,0 +1,153 @@
+import { extend } from '../../../core/utils/extend';
+import { ConfigContext } from '../../../renovation/common/config_context';
+
+interface ComponentProps {
+  className?: string;
+  itemTemplate?: string | (() => string | HTMLElement);
+  valueChange?: EventCallback<any>;
+}
+
+export interface DomComponentWrapperProps {
+  componentType: ComponentClass<any>;
+  templateNames: string[];
+  componentProps: ComponentProps;
+}
+
+export class DomComponentWrapper extends InfernoComponent<DomComponentWrapperProps> {
+  public state: any = {};
+
+  public refs: any = null;
+
+  private readonly widgetRef: RefObject<HTMLDivElement> = createRef();
+
+  private instance: DomComponent | null = null;
+
+  private prevProps: ComponentProps | null = null;
+
+  constructor(props) {
+    super(props);
+    this.getInstance = this.getInstance.bind(this);
+    this.setupWidget = this.setupWidget.bind(this);
+    this.updateWidget = this.updateWidget.bind(this);
+  }
+
+  getConfig(): any {
+    const { id } = (ConfigContext as any);
+    if (this.context[id]) {
+      return this.context[id];
+    }
+    return (ConfigContext as any).defaultValue;
+  }
+
+  render(): JSX.Element {
+    return (
+      <div
+        ref={this.widgetRef}
+        className={this.props.componentProps.className}
+        {...this.props}
+      />
+    );
+  }
+
+  componentWillUpdate(nextProps: DomComponentWrapperProps, nextState, context): void {
+    super.componentWillUpdate(nextProps, nextState, context);
+  }
+
+  createEffects(): InfernoEffect[] {
+    return [
+      new InfernoEffect(this.setupWidget, []),
+      new InfernoEffect(
+        this.updateWidget,
+        [
+          this.props.componentProps,
+          this.getConfig(),
+          this.props.templateNames,
+        ],
+      )];
+  }
+
+  updateEffects(): void {
+    const dependency = [this.props.componentProps, this.getConfig(), this.props.templateNames];
+    this._effects[1]?.update(dependency);
+  }
+
+  setupWidget(): DisposeEffectReturn {
+    const current = this.widgetRef.current as HTMLDivElement;
+    // eslint-disable-next-line new-cap
+    const componentInstance = new this.props.componentType(current, this.getProperties());
+    this.instance = componentInstance;
+    return () => {
+      componentInstance.dispose();
+      this.instance = null;
+    };
+  }
+
+  updateWidget(): void {
+    if (!this.instance) {
+      return;
+    }
+    const updatedOptions = getUpdatedOptions(this.prevProps ?? {}, this.getProperties());
+    if (updatedOptions.length) {
+      this.instance.beginUpdate();
+      updatedOptions.forEach((_ref2) => {
+        const {
+          path,
+          value,
+        } = _ref2;
+        this.instance?.option(path, value);
+      });
+      this.instance.endUpdate();
+    }
+    this.prevProps = this.getProperties();
+  }
+
+  getProperties(): any {
+    const normalizedProps = normalizeProps(this.props.componentProps);
```

Comment:
in non-migrated `normalizeProps` is a local function, not imported from `inferno`

Reply:
I think it's just a coincidence that we have same name of inferno util, and local function
All other renovation files don't have such local function
Let's also ask @williamvinogradov

Reply:
Discussed with @williamvinogradov, there are two different 'normalizeProps': from inferno, and from local func in this file. We should distinguish them and use inferno's in compiled code, and local's where it was used locally before generation